### PR TITLE
Show dishes by categories

### PR DESCRIPTION
**Pull request by cp-dharti-r**
implement-category-dishes

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `RestaurantCategory` component for fetching and displaying category-specific data.
  - Updated restaurant menu pages to include categories alongside dishes.

- **Refactor**
  - Moved variable declarations to local scopes in multiple components to prevent data fetching issues.
  - Enhanced cuisine and food category components with improved key management for better performance.

- **Style**
  - Updated Tailwind CSS breakpoints for better responsiveness across devices.

- **Bug Fixes**
  - Addressed key reassignment in multiple components to ensure unique and accurate rendering of lists.

- **Documentation**
  - Updated component props and types to reflect new features and refactors, ensuring clarity in code maintenance and future development.

### CODE REVIEW

File: website/components/FoodCategory/index.tsx [hunk 1]
```diff
@@ -15,10 +15,10 @@ import "swiper/css";
 import "swiper/css/navigation";
 import CategorySwiperSkeleton from "../SkeletonPlaceholders/CategorySwiper";
 
-// Manage twice data calling issue using this flag
-let dataCalled: any = false;
-
 const FoodCategory = () => {
+  // Manage twice data calling issue using this flag
+  let dataCalled: any = false;
```
Comment (nidhi-canopas) [inline website/components/FoodCategory/index.tsx]: Should it be manage based on the retrieved data is empty or not? It's not the best practice to manage based on flag.

File: website/next.config.mjs [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  experimental: {
```
Comment (nidhi-canopas) [inline website/next.config.mjs]: Let's make sure it's advised to use for production.

File: website/components/FoodCategory/index.tsx (hunk 1 above)
Comment (cp-dharti-r) [reply website/components/FoodCategory/index.tsx]: yaa we can do it but the state management is taking time to set data and meanwhile it's calling twice, that why added flag to manage it smoothly.
inshort it's calling twice in micro second that's why I need to manage it using hard thing

Comment (nidhi-canopas) [reply website/components/FoodCategory/index.tsx]: How do we call it? While mounting the component?

Comment (cp-dharti-r) [reply website/components/FoodCategory/index.tsx]: yaa

Comment (nidhi-canopas) [reply website/components/FoodCategory/index.tsx]: then it should be called once only i guess.

Comment (cp-dharti-r) [reply website/components/FoodCategory/index.tsx]: hmm, but it's calling twice

Comment (nidhi-canopas) [reply website/components/FoodCategory/index.tsx]: I think we should investigate what is causing recalling, instead of preventing it with flag.

**Review by nidhi-canopas — APPROVED** [review]

**Pull request merged**